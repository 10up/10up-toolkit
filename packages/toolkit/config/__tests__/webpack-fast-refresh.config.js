import webpackSerializer from '../../test-utils/webpack-serializer';

describe('webpack.config.js', () => {
	beforeAll(() => {
		expect.addSnapshotSerializer(webpackSerializer);
	});

	it('returns proper configs for project configs', () => {
		let webpackConfig;
		jest.isolateModules(() => {
			// eslint-disable-next-line global-require
			webpackConfig = require('../webpack-fast-refresh.config');
		});

		expect(webpackConfig).toMatchSnapshot();
	});
});
