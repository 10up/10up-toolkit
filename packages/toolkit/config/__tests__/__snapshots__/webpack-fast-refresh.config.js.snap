// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack.config.js returns proper configs for project configs 1`] = `
Object {
  "entry": Object {
    "react-refresh-runtime": Object {
      "import": "react-refresh/runtime.js",
      "library": Object {
        "name": "ReactRefreshRuntime",
        "type": "window",
      },
    },
  },
  "mode": "development",
  "name": "react-refresh-runtime",
  "output": Object {
    "filename": "[name]/index.min.js",
    "path": "/dist/fast-refresh",
  },
  "plugins": Array [
    DependencyExtractionWebpackPlugin: {"combineAssets":false,"combinedOutputFile":null,"externalizedReport":false,"injectPolyfill":false,"outputFormat":"php","outputFilename":null,"useDefaults":false},
  ],
}
`;
